search/features_layer_path_finder.hpp

```diff
@@ -59,42 +59,50 @@ class FeaturesLayerPathFinder
 // TODO (@y): remove following code as soon as
 // FindReachableVertices() will work fast for most cases
 // (significantly less than 1 second).
-#if defined(DEBUG)
+#ifdef DEBUG
     for (auto const * layer : layers)
-      LOG(LINFO, (DebugPrint(*layer)));
-    base::Timer timer;
-#endif  // defined(DEBUG)
-
-    std::vector<IntersectionResult> results;
-    FindReachableVertices(matcher, layers, results);
+      LOG(LDEBUG, (DebugPrint(*layer)));
 
-#if defined(DEBUG)
-    LOG(LINFO, ("Found:", results.size(), "elapsed:", timer.ElapsedSeconds(), "seconds"));
-#endif  // defined(DEBUG)
-
-    for_each(results.begin(), results.end(), std::forward<TFn>(fn));
+    size_t count = 0;
+    base::Timer timer;
+#endif  // DEBUG
+
+    FindReachableVertices(matcher, layers, [&](IntersectionResult const & r)
+    {
+      fn(r);
+#ifdef DEBUG
+      ++count;
+#endif  // DEBUG
+    });
+
+#ifdef DEBUG
+    LOG(LDEBUG, ("Found:", count, "elapsed:", timer.ElapsedSeconds(), "seconds"));
+#endif  // DEBUG
   }
 
   static void SetModeForTesting(Mode mode) { m_mode = mode; }
 
 private:
   void BailIfCancelled() { ::search::BailIfCancelled(m_cancellable); }
 
+  template <class FnT>
   void FindReachableVertices(FeaturesLayerMatcher & matcher,
                              std::vector<FeaturesLayer const *> const & layers,
-                             std::vector<IntersectionResult> & results);
+                             FnT &&fn);
```

Comment:
```suggestion
                             FnT && fn);
```
И в других местах ниже тоже, не по стилю же.